subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r26/TestLauncherCrossVersionSpec.groovy

```diff
@@ -148,6 +150,7 @@ class TestLauncherCrossVersionSpec extends TestLauncherSpec {
         assertTaskNotExecuted(":test")
     }
 
+    @IgnoreIf({ GradleContextualExecuter.embedded})
```

Comment:
Why did this work, but not anymore, in embedded mode?

Reply:
We have problems loading the file system watching library when starting a Gradle build via the tooling API in debug (= embedded) mode. That is why we need to exclude the tests. If I recall correctly the problem there is that Gradle then tries to load the native library in two different classloaders in the same JDK, which isn't allowed. I think this limitation is fine, since it doesn't affect any real builds.

Reply:
It would be good to capture this somewhere for later reference. A comment here would be okay I think, but it seems we bump into this limitation in other places as well. Would perhaps be better to document the limitation in one place, then link to it from these locations.

Reply:
I extracted a method.